Tasks/AzureResourceGroupDeployment/main.ts

```diff
@@ -26,13 +26,9 @@ function run(): Promise<void> {
     }
 }
 
-try {
-    tl.setResourcePath(path.join(__dirname, "task.json"));
-}
-catch (err) {
-    tl.setResult(tl.TaskResult.Failed, tl.loc("TaskNotFound", err));
-    process.exit();
-}
+var taskManifestPath = path.join(__dirname, "task.json");
+tl.debug("Setting resource path to " + taskManifestPath);
+tl.setResourcePath(taskManifestPath);
 
 run().then((result) =>
```

Comment:
TaskNotFound should not needed to be caught at runtime. It should always work. At the best, it should be a debug.assert, but currently we do not have debug.assert in task lib. I will forward you Bryan's mail where he talked about this.